dss/api/subscriptions.py

```diff
@@ -3,35 +3,169 @@
 from werkzeug.exceptions import BadRequest
 
-from .. import get_logger
-from .. import DSS_ELASTICSEARCH_INDEX_NAME, DSS_ELASTICSEARCH_DOC_TYPE, DSS_ELASTICSEARCH_QUERY_TYPE
-from .. import DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME, DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE
+from .. import (get_logger,
+                DSS_ELASTICSEARCH_INDEX_NAME, DSS_ELASTICSEARCH_DOC_TYPE, DSS_ELASTICSEARCH_QUERY_TYPE,
+                DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME, DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE)
 from ..blobstore import BlobNotFoundError
 from ..config import Config
+from ..error import DSSException, dss_handler
 from ..hcablobstore import FileMetadata, HCABlobStore
 from ..util.es import ElasticsearchClient, get_elasticsearch_index
 
 logger = get_logger()
 
 
+@dss_handler
 def get(uuid: str, replica: str):
-    pass
+    owner = request.token_info['email']
 
+    es_client = ElasticsearchClient.get(logger)
 
+    try:
+        response = es_client.get(index=DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME,
+                                 doc_type=DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE,
+                                 id=uuid)
+    except elasticsearch.exceptions.NotFoundError as ex:
+        raise DSSException(requests.codes.not_found, "not_found", "Cannot find subscription!")
+
+    source = response['_source']
+    source['uuid'] = uuid
+    source['replica'] = replica
+
+    if source['owner'] != owner:
+        raise DSSException(requests.codes.forbidden, "forbidden", "Your credentials can't access this subscription!")
+
+    return jsonify(source), requests.codes.okay
+
+
+@dss_handler
 def find(replica: str):
-    pass
+    owner = request.token_info['email']
+    es_client = ElasticsearchClient.get(logger)
+
+    search_obj = Search(using=es_client,
+                        index=DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME,
+                        doc_type=DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE)
+    search = search_obj.query({'match': {'owner': owner}})
+
+    responses = [{
+        'uuid': hit.meta.id,
+        'replica': replica,
+        'owner': owner,
+        'callback_url': hit.callback_url,
+        'query': hit.query.to_dict()}
+        for hit in search.scan()]
 
+    full_response = {'subscriptions': responses}
+    return jsonify(full_response), requests.codes.okay
 
+
+@dss_handler
 def put(extras: dict, replica: str):
-    pass
+    uuid = str(uuid4())
+    query = extras['query']
+    owner = request.token_info['email']
+
+    es_client = ElasticsearchClient.get(logger)
+
+    index_mapping = {
+        "mappings": {
+            DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE: {
+                "properties": {
+                    "owner": {
+                        "type": "string",
+                        "index": "not_analyzed"
+                    }
+                }
+            }
+        }
+    }
+    get_elasticsearch_index(es_client, DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME, logger, index_mapping)
+
+    try:
```

Comment:
@mikebaumann I'm going under the assumption that an exception is thrown if a document is not added. The example they give in the Elasticsearch documentation (https://www.elastic.co/guide/en/elasticsearch/guide/master/create-doc.html) of a document not being added responds with top-level objects of error and status. Status is 409, which translates nicely to Elasticsearch's Python Bindings (https://elasticsearch-py.readthedocs.io/en/master/exceptions.html) ConflictError. I think it's reasonable to assume Elasticsearch would throw this in this situation and another exception if for any other reason a doc can't be stored, so I'm catching all with ElasticsearchException.